homeassistant/components/ozw/config_flow.py

```diff
@@ -1,24 +1,175 @@
+
+        if not self.hass.components.hassio.is_hassio():
+            return self._async_use_mqtt_integration()
+
+        return await self.async_step_on_supervisor()
+
+    @callback
+    def _async_use_mqtt_integration(self):
+        """Handle logic when using the MQTT integration."""
         if "mqtt" not in self.hass.config.components:
             return self.async_abort(reason="mqtt_required")
+
+        return self.async_create_entry(
+            title=TITLE,
+            data={
+                CONF_USB_PATH: self.usb_path,
+                CONF_NETWORK_KEY: self.network_key,
+                CONF_INTEGRATION_CREATED_ADDON: self.integration_created_addon,
+            },
+        )
+
+    async def async_step_on_supervisor(self, user_input=None):
+        """Handle logic when on Supervisor host."""
         if user_input is not None:
-            return self.async_create_entry(title=TITLE, data={})
+            if user_input[CONF_USE_ADDON]:
+                return await self._async_use_addon()
+            return self._async_use_mqtt_integration()
+
+        return self.async_show_form(
+            step_id="on_supervisor", data_schema=ON_SUPERVISOR_SCHEMA
+        )
+
+    async def _async_use_addon(self):
+        """Handle logic when using the OpenZWave add-on."""
+        if await self._async_is_addon_running():
+            return self._async_use_mqtt_integration()
+
+        if await self._async_is_addon_installed():
+            return await self.async_step_start_addon()
+
+        return await self._async_install_addon()
+
+    async def _async_install_addon(self):
+        """Install add-on."""
+        try:
+            await self.hass.components.hassio.async_install_addon("core_zwave")
+        except HassioAPIError as err:
+            _LOGGER.error("Failed to install OpenZWave add-on: %s", err)
+            return self.async_abort(reason="addon_install_failed")
+        self.integration_created_addon = True
+
+        return await self.async_step_start_addon()
+
+    async def async_step_start_addon(self, user_input=None):
+        """Ask for config and start add-on."""
+        if self.addon_config is None:
+            self.addon_config = await self._async_get_addon_config()
+
+        errors = {}
+
+        if user_input is not None:
+            self.network_key = user_input[CONF_NETWORK_KEY]
+            self.usb_path = user_input[CONF_USB_PATH]
+
+            new_addon_config = {CONF_ADDON_DEVICE: self.usb_path}
+            if self.network_key:
+                new_addon_config[CONF_ADDON_NETWORK_KEY] = self.network_key
+
+            if new_addon_config != self.addon_config:
+                await self._async_set_addon_config(new_addon_config)
+
+            try:
+                await self.hass.components.hassio.async_start_addon("core_zwave")
+            except HassioAPIError as err:
+                _LOGGER.error("Failed to start OpenZWave add-on: %s", err)
```

Comment:
The connection that checks the USB path happens after the add-on has started, so we don't get any feedback if the connection fails and the add-on stops after that.

Reply:
We can just report that we failed to start the add-on and then link them to the add-on page to check the logs for now.

Reply:
Or maybe just a message: "Failed to start the add-on. Please check the logs and verify you have entered the right USB path."

Reply:
Yeah, but the problem is that we don't know that the add-on fails since that's done after the start. The start is successful, then after a couple of seconds the connection is tried and if that fails the add-on stops.

Reply:
Ah I see. I wonder if we should add a webserver to the add-on that we can hit as a health check? Or maybe we can ping the docker host. Anyway, let's keep that for a future PR 👍

Reply:
It would also be great if you didn't need to enter a USB path at all, and instead the add-on just probed the serial ports to determine which one is a Z-Wave stick. Is that at all possible? If so I'd love to write up a PR…

Reply:
Yeah, it's something we want to do. We want to take it one step further actually. You plug in a stick -> we start the config flow of the integration that supports it. 

ZHA already supports the available ports: https://github.com/home-assistant/core/blob/dev/homeassistant/components/zha/config_flow.py#L42

But that's for a future PR too. Small steps! 

We have some more plans for Z-Wave onboarding. @MartinHjelmare when you have time, let's create some issues for the pending work so @adamjernst and others can help.

Reply:
I would much appreciate it — anything that can be sliced off, I'm highly motivated to help 👍